src/Compilers/CSharp/Portable/NullableReferenceFlags.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    [Flags]
+    internal enum NullableReferenceFlags
```

Comment:
If I understood correctly, for purpose of experimenting with prototype:
MadsMode for 8.0 is AllowAssemblyOptOut
MadsMode for 8.0+NN is IncludeNonNullWarnings+AllowAssemblyOptOut
NealsMode is IncludeNonNullWarnings+InferLocalNullability+AllowMemberOptOut